crates/sui-types/src/transaction.rs

```diff
@@ -1490,6 +1490,12 @@ impl VersionedProtocolMessage for TransactionData {
             });
         }
 
+        if !protocol_config.random_beacon() && self.uses_randomness() {
```

Comment:
why does this now go in `check_version_supported`? It doesn't seem to have anything to do with the message version?

Reply:
This function doesn't actually just check message version. The next line checks a bunch of feature flags and whether some of the transaction features used are enabled by the config.

Reply:
nit, but while you're in the area, maybe rename it to something more representative to what it currently does?